src/openrct2/core/FileScanner.cpp

```diff
@@ -83,32 +83,28 @@ class FileScannerBase : public IFileScanner
         _recurse = recurse;
         _patterns = GetPatterns(Path::GetFileName(pattern));
 
-        _currentPath = Memory::Allocate<utf8>(MAX_PATH);
-        _currentFileInfo = Memory::Allocate<FileInfo>();
+        _currentPath.resize(MAX_PATH);
 
         Reset();
     }
 
     ~FileScannerBase() override
     {
-        Memory::Free(_currentPath);
-        Memory::Free(_currentFileInfo);
     }
 
     const FileInfo* GetFileInfo() const override
```

Comment:
Didn't @ZehMatt  mean:
```suggestion
    const FileInfo& GetFileInfo() const override
```